### PR TITLE
Add example app using Flask

### PR DESCRIPTION
**Pull request by JamieDanielson**
## Which problem is this PR solving?

- Add example app using Flask (initial plan to migrate from examples repo but the gatekeeper example is much more involved; this is a more basic example)

## Short description of the changes

- Add a few more items to `.gitignore` for virtual environment and other local files
- Add example app with Flask; auto-instrumentation and manual span with trace context

### CODE REVIEW

File: examples/flask/pyproject.toml [hunk 1]
```diff
@@ -6,7 +6,6 @@ authors = ["honeycombio"]
 
 [tool.poetry.dependencies]
 python = "^3.8"
-# honeycomb-beeline = "^2.17.2"
-honeycomb-beeline = { path = "../../../beeline-python" }
+# honeycomb-beeline = "^2.17.2" # uncomment if not using local beeline
```
Comment (JamieDanielson) [inline examples/flask/pyproject.toml:9]: I am not sure actually if it's using the local beeline or if it's using the published beeline...

File: examples/flask/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+```
+
+The server will start listening on port 5000:
+
+```bash
+$ curl localhost:5000
+Hello World
+```
+
+Check out results in Honeycomb! To [deactivate the virtual environment](https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/#leaving-the-virtual-environment):
```
Comment (MikeGoldsmith) [inline examples/flask/README.md:39]:
```suggestion
Check out the results in Honeycomb!

To [deactivate the virtual environment](https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/#leaving-the-virtual-environment):
```

**Comment by MikeGoldsmith** [comment]
@JamieDanielson I updated the example's dependency to use a local path and the README to use poetry as you'd created the pyproject.toml -- I hope that's okay 😄

**Review by MikeGoldsmith — APPROVED** [review]
Looks good 👍🏻

**Pull request merged**